strutl.cc: the filename can't be longer than 255

The URItoFileName translates the path into the filename, but the
filename can't be longer than 255 according to
/usr/include/linux/limits.h.

Truncate it when it is longer than 240 (leave some spaces for
".Packages" and "._Release" suffix)

Upstream-Status: Pending
Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 apt-pkg/contrib/strutl.cc |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/apt-pkg/contrib/strutl.cc b/apt-pkg/contrib/strutl.cc
--- a/apt-pkg/contrib/strutl.cc
+++ b/apt-pkg/contrib/strutl.cc
@@ -399,7 +399,12 @@ string URItoFileName(const string &URI)
    // "\x00-\x20{}|\\\\^\\[\\]<>\"\x7F-\xFF";
    string NewURI = QuoteString(U,"\\|{}[]<>\"^~_=!@#$%^&*");
    replace(NewURI.begin(),NewURI.end(),'/','_');
-   return NewURI;
+
+   // Truncate from the head when it is longer than 240
+   if(NewURI.length() > 240)
+       return NewURI.substr(NewURI.length() - 240, NewURI.length() - 1);
+   else
+       return NewURI;
 }
 									/*}}}*/
 // Base64Encode - Base64 Encoding routine for short strings		/*{{{*/
-- 
1.7.10.4

